Content.Server/Administration/Logs/AdminLogManager.Json.cs

```diff
@@ -26,61 +26,101 @@ private void InitializeJson()
         foreach (var converter in _reflection.FindTypesWithAttribute<AdminLogConverterAttribute>())
         {
             var instance = _typeFactory.CreateInstance<JsonConverter>(converter);
-            (instance as IAdminLogConverter)?.Init(_dependencies);
             _jsonOptions.Converters.Add(instance);
         }
 
         var converterNames = _jsonOptions.Converters.Select(converter => converter.GetType().Name);
         _sawmill.Debug($"Admin log converters found: {string.Join(" ", converterNames)}");
     }
 
-    private (JsonDocument Json, HashSet<Guid> Players, List<AdminLogEntity> Entities) ToJson(
+    private (JsonDocument Json, List<AdminLogPlayer> Players, List<AdminLogEntity> Entities) ToJson(
+        int id,
         Dictionary<string, object?> properties)
     {
-        var entities = new Dictionary<EntityUid, AdminLogEntity>();
-        var players = new HashSet<Guid>();
-        var parsed = new Dictionary<string, object?>();
+        var entities = new List<AdminLogEntity>();
+        var players = new List<AdminLogPlayer>();
+        var parsed = new Dictionary<string, object?>(properties.Count);
```

Comment:
These two collections could be pooled using an ObjectPool. According to https://learn.microsoft.com/en-us/aspnet/core/performance/objectpool?view=aspnetcore-7.0, they are thread safe.
Parsed can be returned at the end of this method after it is serialized into json.
Players can be returned once it is read outside this method. Both ToJson and the Add method called afterwards are private, so this should also be safe.